bin/makePsfLibrary.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python
+#
+# LSST Data Management System
+# Copyright 2015 AURA/LSST
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+import pyfits
+import argparse
+import sys
+import os.path
+import numpy
+import lsst.daf.persistence as dafPersist
+import lsst.afw.cameraGeom as cameraGeom
+import lsst.afw.coord as afwCoord
+import lsst.afw.detection as afwDetection
+import lsst.afw.geom as afwGeom
+import lsst.afw.image as afwImage
+import lsst.afw.math as afwMath
+import lsst.afw.table as afwTable
+import lsst.meas.base as measBase
+import lsst.meas.algorithms as measAlg
+
+psfSize = 33
+rafts = ["R01","R02","R03","R10","R11","R12","R13","R14","R20","R21","R22","R23","R24","R30","R31","R32","R33","R34","R41","R42","R43"]
+
+"""
+    Script to create a library of Psfs from an lsst focal plane.
+
+    @param[in]  catalog          A par file from PhoSim which lists all the stars
+    @param[in]  input_dir        Directory containing the fits.gz files from PhoSim
+    @param[in]  basename         prefix for the fits.gz files, excluding Rnn_Smm.fits.gz
+    @param[in]  output_dir       Directory in which to place the output libraries.
+
+    The output index is placed in output_dir/psfs.index
+    The library is written to output_dir/basename/psfs/psf_library_nn.fits
+    where nn is a two digit form of the raft number
+"""
+
+#   This routine takes a list of images and a catalog name (presumably covering the
+#   same area as the images.  Any entry in the catalog which is fully contained in an
+#   image is cutout, transformed to the center of the pupil, and then placed in a libray
+#   (multi-extension fits file).  The entry is also logged to an open file handle, fout
+def makeSubLibrary(images, catname, output_file, fout):
+    # Create an XYTransform from TanPixels to Pixels.  This is the transform
+    # needed to remove the optical distortion from the Psf. Use R22_S11 as center
+    butler = dafPersist.Butler(root=os.path.join(os.path.dirname(__file__), "."))
```

Comment:
Is hardcoding this path to the location of this file something only appropriate for the example you've been tinkering with, or is this package intended to always be checked out into the work directory.